diesel/src/macros/mod.rs

```diff
@@ -401,9 +401,7 @@ macro_rules! __diesel_table_impl {
         table_doc = [$($doc:expr,)*];
         $($rest:tt)*
     ) => {
-        // Remove this workaround as soon as rust issue 40872 is implemented
-        // https://github.com/rust-lang/rust/issues/40872
-        const ERROR: i32 = env!("invalid table! syntax");
+        compile_error!("invalid table! syntax");
```

Comment:
Can we link to some docs here? (If we have them)

Reply:
Should that link appear in the error message or a comment just above the code?

Reply:
in the error message, so users can see how to use `table!` correctly